### ISSUE TITLE
Failed to disable `delegateEvents` after upgrading babel-preset-solid from 0.23.8 to 0.24.0

### ISSUE THREAD

**Issue by zhaoyao91** [ryansolid/solid#334]
**Describe the bug**

Before upgrading, I use configured babel-preset-solid as below

```
    [
      "babel-preset-solid",
      {
        "delegateEvents": false
      }
    ]
```

It will disable the events delegation correctly.

But after upgrading, the same config does not take any effect to disable the event delegation.

**To Reproduce**

- create a solid app with babel-preset-solid configured as

```
    [
      "babel-preset-solid",
      {
        "delegateEvents": false
      }
    ]
```
- create some component and register some event handler such as onClick
- check the devtool to see if there is a global event delegation handler on document.

**Expected behavior**

- no such global handler on document

**Additional context**

solid-js is also updated from 0.23.11 to 0.24.0

**Comment by ryansolid** [comment]
Yeah it was removed it since it doesn't make sense because as pointed out it would inconsistent between runtime and compiler that isn't exactly reconcilable (I can think of a very manual way to do so but it seemed unnecessary). The motivation was the behavior was changed now that only specific events are delegated, specifically input events that are composed and bubble. 

Is there a reason you want to remove event delegation?

**Comment by zhaoyao91** [comment]
I am in a situation where I need to disable event delegation.

We are building async modules which could be loaded in a app. Each module is an isolated js bundle with a version of solid in it. Currently every module's solid will attach a handler to the doucment which lead to many duplicate handlers in the document, and which lead to unexpected behaviors (such as one click triggers many clicks of a button).

By disable event delegation, every module will only attach handlers to its only elements, without any global side effect.

**Comment by ryansolid** [comment]
Interesting I suppose it's an argument for React 17's solution. That being said I can add this back to the compiler pretty easily. To prevent it delegating at runtime like in spreads you need to clear the list in each build as well.

```js
import { DelegatedEvents } from "solid-js/web";

DelegatedEvents.clear();
```

**Comment by zhaoyao91** [comment]
Cool.

To conclude, after upgrading of the compiler, we need to do 2 things:

- config `babel-preset-solid` to disable `delegateEvents` for buildtime
- run `DelegatedEvents.clear();` before rendering app for runtime

am I missing something?

**Comment by ryansolid** [comment]
That should be it. I've added back now in the latest version of the babel plugin. If you blow out your lock file it should install it without me doing a new Solid release just yet. (v0.25.1 of babel-plugin-jsx-dom-expressions). Enjoy.

**Comment by zhaoyao91** [comment]
There was some problem with `DelegatedEvents.clear();`

I made a button and pass an onClick handler to it as props spreading. After calling `DelegatedEvents.clear();`, I can found a click event handler on the button element, but there is no effect on actually click it.

![image](https://user-images.githubusercontent.com/3808838/107013838-dc7ff380-67d5-11eb-89ee-bebd0c2b22e0.png)

**Comment by ryansolid** [comment]
Ok fixed in 0.24.1
